packages/grid/x-data-grid/src/models/params/gridCellParams.ts

```diff
@@ -71,51 +64,41 @@ export interface GridCellParams<
 /**
  * GridCellParams containing api.
  */
-export interface GridRenderCellParams<
-  V = any,
-  R = any,
-  F = V,
-  Api extends GridApiCommon = GridApiCommunity,
-> extends GridCellParams<V, R, F, Api> {
+export interface GridRenderCellParams<V = any, R = any, F = V> extends GridCellParams<V, R, F> {
   /**
    * GridApi that let you manipulate the grid.
+   * @deprecated Use the `apiRef` returned by `useGridApiContext` or `useGridApiRef`
```

Comment:
Community users don't have access to `useGridApiRef`.

```suggestion
   * @deprecated Use the `apiRef` returned by `useGridApiContext` or `useGridApiRef` (only available in DataGridPro).
```

x other params.

Reply:
I think that if we set the default value as `any`, 99% of our users will never use it.
And in that case, we better get rid of the whole complexity.

Fixed for the comment.

We have a few use-case where removing the `api` param breaks internal code
For instance in `GRID_CHECKBOX_SELECTION_COL_DEF`:

```ts
  valueGetter: (params) => {
    const selectionLookup = selectedIdsLookupSelector(
      (params as GridValueGetterFullParams).api.state,
      (params as GridValueGetterFullParams).api.instanceId,
    );
    return selectionLookup[params.id] !== undefined;
  },
```

Maybe we will have to pass the `state` and `instanceId` as a param but it would re-introduce generics to type correctly.
But let's take the time to discuss it before v6, the deprecation is for the long term.